### PR TITLE
Miscellaneous documentation refinements and additions.

### PR DESCRIPTION
**Pull request by metasim**
- [X] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [X] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Documentation refinements. Trimming down root content and starting to push specifics down into modules.

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
Thank you for all the documentation.

In the PR that was merged, you wrote "_Rust implementations of [GDAL Programs](https://gdal.org/programs/index.html)_" at
https://github.com/metasim/gdal/blob/7f382ea24b7efffab0f41c7a99612c227560d597/src/programs/mod.rs#L1.

But we just wrote wrappers and did not implement those programs. Maybe we should adapt this.

File: gdal-sys/src/lib.rs [hunk 1]
```diff
@@ -2,6 +2,6 @@
 #![allow(non_camel_case_types)]
 #![allow(non_snake_case)]
 #![allow(clippy::upper_case_acronyms)]
-
+#![allow(rustdoc::bare_urls)]
```
Comment (lnicola) [inline gdal-sys/src/lib.rs:5]: Is there any reason to want this?

File: src/driver.rs [hunk 2]
```diff
@@ -53,38 +102,63 @@ impl Driver {
...
+    /// See also: [`short_name`](Self::short_name`).
     pub fn long_name(&self) -> String {
         let rv = unsafe { gdal_sys::GDALGetDriverLongName(self.c_driver) };
         _string(rv)
     }
 
+    /// Create a new dataset of size (`size_x`, `size_y`) and `bands` band count,
+    /// and [`u8`] as the cell data type.
+    ///
+    /// To specify an alternative data type (e.g. [`f32`]), use [`create_with_band_type`](Self::create_with_band_type).
```
Comment (lnicola) [inline src/driver.rs:144]: TBH, I don't think `u8` deserve special-casing.

File: src/driver.rs [hunk 3]
```diff
@@ -106,6 +199,37 @@ impl Driver {
         self.create_with_band_type_with_options::<T, _>(filename, size_x, size_y, bands, &options)
     }
 
+    /// Create a new dataset of size (`size_x`, `size_y`) and `bands` band count,
+    /// with cell data type specified by `T` and extended options specified via `options`.
+    /// [Per GDAL](https://gdal.org/api/gdaldriver_cpp.html#_CPPv4N10GDALDriver6CreateEPKciii12GDALDataType12CSLConstList),
+    /// the set of legal options for `options` is driver specific, and there is no way to query in advance to establish legal values.a
```
Comment (lnicola) [inline src/driver.rs:205]:
```suggestion
    /// the set of legal options for `options` is driver specific, and there is no way to query in advance to establish the valid ones.
```

File: src/lib.rs [hunk 4]
```diff
@@ -2,176 +2,101 @@
...
+//!
+//! Metadata in GDAL takes a number of forms, some of which are specific to purpose
+//! (e.g. pixel interpretation, spatial reference system),
+//! and other more general-purpose (e.g. acquisition date-time). The former will be covered in
+//! relevant sections of the [`raster`] and [`vector`] modules, and the general-purpose data model
+//! in the [`Metadata`] API.
+//!
+//! ### Raster Data
+//!
+//! A raster `Dataset` has a `size` (`cols`/`rows`), an ordered sequence of [`RasterBand`](raster::RasterBand)s, geospatial
```
Comment (lnicola) [inline src/lib.rs:85]:
```suggestion
//! A raster `Dataset` has a `size` (`cols`, `rows`), an ordered sequence of [`RasterBand`](raster::RasterBand)s, geospatial
```

File: src/programs/mod.rs [hunk 5]
```diff
@@ -1,3 +1,3 @@
-//! Rust implementations of [GDAL Programs](https://gdal.org/programs/index.html)
+//! Rust Wrappers of [GDAL Programs](https://gdal.org/programs/index.html)
```
Comment (lnicola) [inline src/programs/mod.rs:1]:
```suggestion
//! Rust wrappers for the [GDAL Programs](https://gdal.org/programs/index.html)
```

**Review by lnicola — APPROVED** [review]

**Comment by lnicola** [comment]
bors d+

File: src/driver.rs (hunk 2 above)
Comment (metasim) [reply src/driver.rs:144]: @lnicola Is this a comment about the markup I used, or about the existence of a special case `create` function, specifically for `u8`?

Comment (lnicola) [reply src/driver.rs:144]: About the existence of the function, there's nothing we can do about it in this PR.

Comment (metasim) [reply src/driver.rs]: Yeh, I completely agree. It was actually confusing to me when I first came upon it.

**Pull request merged**